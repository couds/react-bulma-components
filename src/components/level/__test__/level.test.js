import React from 'react';
import renderer from 'react-test-renderer';
import Level from '..';

describe('Level component', () => {
  it('Should exist', () => {
    expect(Level).toMatchSnapshot();
  });
  it('Should expose Level Side and Item', () => {
    expect(Level.Side).toMatchSnapshot();
    expect(Level.Item).toMatchSnapshot();
  });
  it('Should have level classname', () => {
    const component = renderer.create(
      <Level>
        <Level.Item>Item 1</Level.Item>
        <Level.Item>Item 2</Level.Item>
        <Level.Item>Item 3</Level.Item>
      </Level>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Level className="other-class this-is-a-test">
        <p>Default</p>
      </Level>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Level style={{ height: 250 }}>
        <p>Default</p>
      </Level>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render Levels with right and left side', () => {
    const component = renderer.create(
      <Level>
        <Level.Side>
          <Level.Item>Item 1</Level.Item>
          <Level.Item>Item 2</Level.Item>
        </Level.Side>
        <Level.Side align="right">
          <Level.Item>Item 3</Level.Item>
        </Level.Side>
      </Level>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have is-mobile class if mobile prop is true', () => {
    const component = renderer.create(
      <Level mobile>
        <Level.Side>
          <Level.Item>Item 1</Level.Item>
          <Level.Item>Item 2</Level.Item>
        </Level.Side>
        <Level.Side align="right">
          <Level.Item>Item 3</Level.Item>
        </Level.Side>
      </Level>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
