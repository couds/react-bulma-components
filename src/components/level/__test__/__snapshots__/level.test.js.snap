// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level component Should concat classname in props with Bulma classname 1`] = `
<div
  className="level other-class this-is-a-test"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Level component Should exist 1`] = `[Function]`;

exports[`Level component Should expose Level Side and Item 1`] = `[Function]`;

exports[`Level component Should expose Level Side and Item 2`] = `[Function]`;

exports[`Level component Should have level classname 1`] = `
<div
  className="level"
  style={Object {}}
>
  <div
    className="level-item"
    style={Object {}}
  >
    Item 1
  </div>
  <div
    className="level-item"
    style={Object {}}
  >
    Item 2
  </div>
  <div
    className="level-item"
    style={Object {}}
  >
    Item 3
  </div>
</div>
`;

exports[`Level component Should render Levels with right and left side 1`] = `
<div
  className="level"
  style={Object {}}
>
  <div
    className="level-left"
    style={Object {}}
  >
    <div
      className="level-item"
      style={Object {}}
    >
      Item 1
    </div>
    <div
      className="level-item"
      style={Object {}}
    >
      Item 2
    </div>
  </div>
  <div
    className="level-right"
    style={Object {}}
  >
    <div
      className="level-item"
      style={Object {}}
    >
      Item 3
    </div>
  </div>
</div>
`;

exports[`Level component Should use inline styles 1`] = `
<div
  className="level"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</div>
`;
